src/quantum/setup.py

```diff
@@ -34,6 +34,7 @@
 ]
 
 DEPENDENCIES = [
+    'azure-storage-blob~=12.14.1'
```

Comment:
Hi @evelyn-ys, can an extension dependence on a specific version of 'azure-storage-blob'?

Reply:
we don't recommend one extension to depend on another one....

Why does `quantum` need to use `azure-storage-blob`?

Reply:
It is not depending on another extension.  It is depending on a published PyPi package, [azure-storage-blob](https://pypi.org/project/azure-storage-blob/) from the [Azure SDK for Python](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/storage/azure-storage-blob/azure/storage/blob).

Why?
To submit a job to quantum hardware or a simulator, the client of the quantum service must upload a quantum program to a blob in an Azure storage account. In the past, there was only one method of submitting a quantum job with the CLI: Create a Q# program and use the Q# compiler.  The compiler builds a .NET executable that performs the upload.  To generalize CLI job-submission to support many other quantum programming languages in addition to Q#, the quantum CLI extension needs to perform this blob upload. 

The Quantum QDK provides an SDK called qdk-python that aids in managing Azure Quantum programming with a Python client.  The blob-upload code being added to the CLI extension in this PR is based on code in qdk-python.  The qdk-python code depends on the azure-storage-blob PyPi package.

Reply:
Other extensions have dependencies on various PyPi packages, for example:

src\cloudservice\setup.py
    'azure-mgmt-compute~=20.0.0'

src\connectedk8s\setup.py
    'kubernetes==24.2.0',
    'pycryptodome==3.14.1',
    'azure-mgmt-hybridcompute==7.0.0'

src\db-up\setup.py
    'Cython==0.29.17',
    'mysql-connector-python==8.0.14',
    'psycopg2-binary==2.9.1'

Reply:
It seems `azure-storage-blob` is not used in azure-cli repo. Approved.